protocol_rfcs/managed-commits.md

```diff
@@ -43,19 +45,44 @@ For example:
-for the information around which one of these are valid. Refer to [managed commits](#managed-commits) for more details.
+The `_delta_log/_commits` directory may contain uncommitted delta files. The [commit-owner](#commit-owner) is the source of truth about which of those delta
+files map to committed versions. Refer to [managed commits](#managed-commits) for more details.
 
-Delta files use new-line delimited JSON format, where every action is stored as a single line JSON document.
+Delta files use newline-delimited JSON format, where every action is stored as a single line JSON document.
 A delta file, corresponding to version `n`, contains an atomic set of [_actions_](#Actions) that should be applied to the previous table state, corresponding to `n-1`, in order to construct the `n`th snapshot of the table.
 An action changes one aspect of the table's state, for example, adding or removing a file.
 
-> ***New Section***
+### Metadata Cleanup
+
+> ***Change to [existing section](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#metadata-cleanup)***
+
+The _delta_log directory grows over time as more and more commits and checkpoints are accumulated.
+Implementations are recommended to delete expired commits and checkpoints in order to reduce the directory size.
+The following steps could be used to do cleanup of the DeltaLog directory:
+1. Identify a threshold (in days) uptil which we want to preserve the deltaLog. Let's refer to
+   midnight UTC of that day as `cutOffTimestamp`. The newest commit not newer than the `cutOffTimestamp` is
+   the `cutoffCommit`, because a commit exactly at midnight is an acceptable cutoff. We want to retain everything including and after the `cutoffCommit`.
+2. Identify the newest checkpoint that is not newer than the `cutOffCommit`. A checkpoint at the `cutOffCommit` is ideal, but an older one will do. Lets call it `cutOffCheckpoint`.
+   We need to preserve the `cutOffCheckpoint` and all commits after it, because we need them to enable
+   time travel for commits between `cutOffCheckpoint` and the next available checkpoint.
+   - **If no checkpoint can be found, do not proceed with metadata cleanup as there is nothing to cleanup.**
+3. Delete all [delta log entries](#delta-log-entries and [checkpoint files](#checkpoints) before the
+   `cutOffCheckpoint` checkpoint. Also delete all the [log compaction files](#log-compaction-files) having
+   startVersion <= `cutOffCheckpoint`'s version.
+4. Now read all the available [checkpoints](#checkpoints-1) in the _delta_log directory and identify
+   the corresponding [sidecar files](#sidecar-files). These sidecar files need to be protected.
+5. List all the files in `_delta_log/_sidecars` directory, preserve files that are less than a day
+   old (as of midnight UTC), to not break in-progress checkpoints. Also preserve the referenced sidecar files
+   identified in Step-4 above. Delete everything else.
```

Comment:
It's probably enough to just show the bullet that changed, e.g.

2. Identify the newest checkpoint that is not newer than the `cutOffCommit`. A checkpoint at the `cutOffCommit` is ideal, but an older one will do. Lets call it `cutOffCheckpoint`.
   We need to preserve the `cutOffCheckpoint` and all commits after it, because we need them to enable
   time travel for commits between `cutOffCheckpoint` and the next available checkpoint.
   - <ins>If no `cutOffCheckpoint` can be found, do not proceed with metadata cleanup as there is nothing to cleanup.</ins>